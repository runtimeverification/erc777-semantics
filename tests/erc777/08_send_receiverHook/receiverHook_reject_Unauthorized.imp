test.preamble();

// send() to other address,  operatorSend() through default operator, operatorSend() from caller.
// receiver hook, always rejecting, sending through unauthorized operator

test.registerTokensReceivedHook(3, 5); //rejecting hook
test.makeCaller(5);
test.operatorSend(10, 3, 10, "", ""); //unauthorized send
