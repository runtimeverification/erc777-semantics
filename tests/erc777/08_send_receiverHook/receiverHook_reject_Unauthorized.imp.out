Registering receiver 3 with hook 5
Changing the caller to 5
5 is sending 10 from 10 to 3 (data:""),  (operatorData:"")
REVERT
Balance of 10 is 100
Balance of 3 is 30
